project/MimaExcludes.scala

```diff
@@ -82,6 +82,10 @@ object MimaExcludes {
             // SPARK-5166 Spark SQL API stabilization
             ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.apache.spark.ml.Transformer.transform"),
             ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.apache.spark.ml.Estimator.fit")
+          ) ++ Seq(
+            // SPARK-5315 Spark Streaming Java API returns Scala DStream
+            ProblemFilters.exclude[MissingMethodProblem](
+              "org.apache.spark.streaming.api.java.JavaDStreamLike.reduceByWindow")
```

Comment:
Hi Patrick, Mima test complains about this, so I added it here.

Besides, what's your opinion about deprecating the old API or modifying it if this API is actually not correct? I'm not sure which is the good way, but @srowen suggested me to deprecate rather than modify.

Thanks a lot :)